### PR TITLE
Add CSI_REMOVE_HOLDER_PODS key and its value in the ocs configmap

### PR DESCRIPTION
**Pull request by iamniting**
set the CSI_REMOVE_HOLDER_PODS to true for new users on version 4.16 onwards and set it to false for older/upgraded user to give them time to migrate manually.

with version 4.17 we will make the CSI_REMOVE_HOLDER_PODS to true for all customers older and new both.

If the configmap is being created for the first time we are considering it as a new install.

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

Jira: https://issues.redhat.com/browse/RHSTOR-5340?focusedId=24357013&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24357013

### CODE REVIEW

**Comment by iamniting** [comment]
@subhamkrai once this PR is merged can you pls read this key from the ocs-operator-config configmap in the rook deployment?

File: controllers/ocsinitialization/ocsinitialization_controller.go
Comment (malayparida2000) [inline controllers/ocsinitialization/ocsinitialization_controller.go:348]: I guess the data section was there do that when the CM is created the data section is not null, will removing this not create any issue?

**Review by malayparida2000 — CHANGES_REQUESTED** [review]
The bundle separation PR https://github.com/red-hat-storage/ocs-operator/pull/2492 is not yet merged. So it will be appropriate to add the change for reading this as env variable to the csv in the csv-merger. This will make sure after that PR is merged all the mentioned env vars are properly migrated to the rook ds repo csv.

File: controllers/ocsinitialization/ocsinitialization_controller.go
Comment (iamniting) [reply controllers/ocsinitialization/ocsinitialization_controller.go:348]: No, it does not create any issues, I have tested it.

**Comment by subhamkrai** [comment]
> @subhamkrai once this PR is merged can you pls read this key from the ocs-operator-config configmap in the rook deployment?

suree

**Comment by iamniting** [comment]
> The bundle separation PR #2492 is not yet merged. So it will be appropriate to add the change for reading this as env variable to the csv in the csv-merger. This will make sure after that PR is merged all the mentioned env vars are properly migrated to the rook ds repo csv.

I am not doing that because if we do so we will have merge conflicts in the other PR where we are removing the rook CSV.

File: controllers/ocsinitialization/ocsinitialization_controller.go
Comment (subhamkrai) [inline controllers/ocsinitialization/ocsinitialization_controller.go:362]:
```suggestion
		} else if ocsOperatorConfig.Data[util.CsiRemoveHolderPodsKey] != "" {
			ocsOperatorConfigData[util.CsiRemoveHolderPodsKey] = ocsOperatorConfig.Data[util.CsiRemoveHolderPodsKey]
```

do we need this else if?

Comment (iamniting) [reply controllers/ocsinitialization/ocsinitialization_controller.go:362]: yes we do, this is the third case where we want to keep the value that was set. If we don't do that it will always remove it.

**Comment by malayparida2000** [comment]
> > The bundle separation PR #2492 is not yet merged. So it will be appropriate to add the change for reading this as env variable to the csv in the csv-merger. This will make sure after that PR is merged all the mentioned env vars are properly migrated to the rook ds repo csv.
> 
> I am not doing that because if we do so we will have merge conflicts in the other PR where we are removing the rook CSV.

That should be okay right? I just want to make sure that all the env vars defined in the csv merger get transferred directly to the rook ds. If we don't want the merge conflict, IMO we should wait till the bundle separation PR merges and then merge this.

File: controllers/ocsinitialization/ocsinitialization_controller.go
Comment (subhamkrai) [reply controllers/ocsinitialization/ocsinitialization_controller.go:362]: since we are setting the same value, where are we removing it if we don't set?

Comment (iamniting) [reply controllers/ocsinitialization/ocsinitialization_controller.go:362]: actually if i do not use this 3rd case the value will be removed as when the variable was created this key was not part of the map and it will be replaced.

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**
